builder/openstack/step_allocate_ip.go

```diff
@@ -15,53 +16,78 @@ type StepAllocateIp struct {
 
 func (s *StepAllocateIp) Run(state multistep.StateBag) multistep.StepAction {
 	ui := state.Get("ui").(packer.Ui)
-	csp := state.Get("csp").(gophercloud.CloudServersProvider)
-	server := state.Get("server").(*gophercloud.Server)
+	config := state.Get("config").(Config)
+	server := state.Get("server").(*servers.Server)
+
+	// We need the v2 compute client
+	client, err := config.computeV2Client()
+	if err != nil {
+		err = fmt.Errorf("Error initializing compute client: %s", err)
+		state.Put("error", err)
+		return multistep.ActionHalt
+	}
 
-	var instanceIp gophercloud.FloatingIp
+	var instanceIp *floatingip.FloatingIP
 	// This is here in case we error out before putting instanceIp into the
 	// statebag below, because it is requested by Cleanup()
 	state.Put("access_ip", instanceIp)
 
 	if s.FloatingIp != "" {
-		instanceIp.Ip = s.FloatingIp
+		*instanceIp = floatingip.FloatingIP{FixedIP: s.FloatingIp}
 	} else if s.FloatingIpPool != "" {
-		newIp, err := csp.CreateFloatingIp(s.FloatingIpPool)
+		newIp, err := floatingip.Create(client, floatingip.CreateOpts{
+			Pool: s.FloatingIpPool,
+		}).Extract()
 		if err != nil {
 			err := fmt.Errorf("Error creating floating ip from pool '%s'", s.FloatingIpPool)
 			state.Put("error", err)
 			ui.Error(err.Error())
 			return multistep.ActionHalt
 		}
-		instanceIp = newIp
-		ui.Say(fmt.Sprintf("Created temporary floating IP %s...", instanceIp.Ip))
+
+		*instanceIp = *newIp
+		ui.Say(fmt.Sprintf("Created temporary floating IP %s...", instanceIp.FixedIP))
 	}
 
-	if instanceIp.Ip != "" {
-		if err := csp.AssociateFloatingIp(server.Id, instanceIp); err != nil {
-			err := fmt.Errorf("Error associating floating IP %s with instance.", instanceIp.Ip)
+	if instanceIp != nil && instanceIp.FixedIP != "" {
```

Comment:
FixedIP is only set if you supplied in IP through `floating_ip`. If an IP is retuned from the pool it will be set in the IP field. This means that the instance will not get its IP allocated if it's fetched from the pool.